clientd3d/draw3d.c

```diff
@@ -533,30 +533,48 @@ void SetLightingInfo(int sun_x, int sun_y, BYTE intensity)
  */
 void StretchImage(void)
 {
-  StretchC1To2(gBits,gBufferBits,area.cx,area.cy);
+  Stretch(gBits, gBufferBits, area.cx, area.cy, main_viewport_width, main_viewport_height, MAXX);
 }
 /************************************************************************/
-void StretchC1To2(BYTE *src,BYTE *dest,int width,int height)
+void Stretch(BYTE* src, BYTE* dest, int width, int height, int new_width, int new_height, int max_width)
 {
-  register long i,j;
-  register BYTE *s,*d,*d2;
-  
-  /* 1->2 stretch */
-  for(i = 0;i < height;i++)
-  {
-    s = src + i * MAXX;
-    d  = dest + 2 * i * MAXX * 2;
-    d2 = d + MAXX * 2;                // One row offset from dest
-    for(j = 0; j < width; j++)
+    // If the new size is exactly the same as the original, perform a direct copy
```

Comment:
It seems like this is extremely unlikely; is it worth bothering with this special case?